processor/src/main/java/org/mapstruct/ap/internal/util/accessor/Accessor.java

```diff
@@ -26,17 +26,39 @@
 import javax.lang.model.type.TypeMirror;
 
 /**
+ * This represents an Accessor that can be used for writing/reading a property to/from a bean.
+ *
  * @author Filip Hrisafov
  */
 public interface Accessor {
 
+    /**
+     * This returns the type that this accessor gives as a return.
+     *
+     * e.g. The {@link ExecutableElement#getReturnType()} if this is a method accessor,
+     * or {@link javax.lang.model.element.VariableElement#asType()} for field accessors.
+     *
+     * @return the type that the accessor gives as a return
+     */
```

Comment:
Not really. If it is a setter this will be `Void`. Maybe I need to rename it to `getReturnType`. Internally, there are different mechanics based on the parameters of the `ExecutableElement` that the `Accessor ` returns.